From 44110a51605032d4a2716df0c73884815c581feb Mon Sep 17 00:00:00 2001
From: Rajesh Bhalerao <Rajesh_Bhalerao@xyratex.com>
Date: Thu, 23 May 2013 17:30:27 +0530
Subject: [PATCH 25/25] If there is m0_sm_timedwait() for RI_SUCCESS or
 RI_FAILURE of rm_incoming, it's not necessary to check the sate of
 rm_incoming (esp. if it's RI_WAIT).

---
 rm/ut/file.c | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/rm/ut/file.c b/rm/ut/file.c
index f900ade..b237d55 100644
--- a/rm/ut/file.c
+++ b/rm/ut/file.c
@@ -230,12 +230,10 @@ static void wait_lock(enum rm_server srv_id)
 
 	m0_file_lock(owner, in);
 	m0_rm_owner_lock(owner);
-	if (incoming_state(in) == RI_WAIT) {
-		rc = m0_sm_timedwait(&in->rin_sm,
-				     M0_BITS(RI_SUCCESS, RI_FAILURE),
-				     M0_TIME_NEVER);
-		M0_UT_ASSERT(rc == 0);
-	}
+	rc = m0_sm_timedwait(&in->rin_sm,
+			     M0_BITS(RI_SUCCESS, RI_FAILURE),
+			     M0_TIME_NEVER);
+	M0_UT_ASSERT(rc == 0);
 	m0_rm_owner_unlock(owner);
 	M0_UT_ASSERT(in->rin_rc == 0);
 	m0_file_unlock(in);
@@ -281,12 +279,10 @@ static void dlock(enum rm_server srv_id, int n)
 	m0_file_lock(owner, &req);
 	m0_rm_owner_lock(owner);
 	test_verify(DISTRIBUTED_LOCK_TEST);
-	if (incoming_state(&req) == RI_WAIT) {
-		rc = m0_sm_timedwait(&req.rin_sm,
-				     M0_BITS(RI_SUCCESS, RI_FAILURE),
-				     M0_TIME_NEVER);
-		M0_UT_ASSERT(rc == 0);
-	}
+	rc = m0_sm_timedwait(&req.rin_sm,
+			     M0_BITS(RI_SUCCESS, RI_FAILURE),
+			     M0_TIME_NEVER);
+	M0_UT_ASSERT(rc == 0);
 	M0_UT_ASSERT(req.rin_rc == 0);
 	M0_UT_ASSERT(incoming_state(&req) == RI_SUCCESS);
 	m0_rm_owner_unlock(owner);
-- 
1.8.3.2

